### DISCUSSION TITLE
16.0

### DISCUSSION THREAD

**Discussion by carloscabanero** [blinksh/blink#1570] [Announcements]
Main feature in this version is WebAuthn keys support, with both as Passkeys and Secure Keys flavor. We continue making fixes for iOS16 and iPadOS16, specially on the keyboard side, and at least for those cases where the ball is on our side. 

## Here is what to try:

- You can use Passkeys for SSH Authentication.
- You can create a WebAuthn key in your Yubikey for SSH Authentication. 
- Other Secure Keys should be supported, please get exotic in your use cases and let us know if there are any issues. We may not able to do much, but we will raise bugs to Apple. For example, we know Titan keys can be created but they cannot sign requests sometimes.

- Key autorepeat should be back on iPadOS16.
- SmarKey should not show up when using a hardware keyboard anymore. There may still be corner cases when connecting / disconnecting the keyboard, but at this point we are unsure if these are all on Apple side. Let us know what you find.

### Build 573 (Sep 23)
- We changed the key type to be sk-ecdsa instead of webauthn-sk-ecdsa
- We added docs for WebAuthn keys, please check them out and let us know if we are missing something or anything is unclear. https://docs.blink.sh/advanced/webauthn

### Build 591 (Oct 10)
- Biggest change on this build. We have reworked the layout code on our Smart Keys bar in order to handle StageManager a whole lot better. And we think we have succeeded at creating a great experience!
    - Now, we need your feedback. Our software keyboard is very important for us.  The new Stage Manager brings a new way to work with multiple apps coexisting at different sizes and positions, and those windows now move from the back to the front all the time. That is where we need your feedback. We like the way Blink is now tracking those changes and tries to maximize what you can see on the screen. But what do you think? Are there layouts where you think Blink should behave better? Let us know!
- Code handling host names with special characters and in a case sensitive manner. Fixes #1440
- Adjusted Smart Keys placement on Extra Space setting for 10.9 keyboard.
- Fixed multi-line prompts on ssh authentication. Issue #1577
- Do not override env LANG. Fixes #1562
- Improved jump hosts handling. Issue #1450 
- Fixed logs not showing key name on rejection. Fixes #1576

### Build 595 (Oct 11)
- Added new iPhone model layouts for three finger taps. 

### Build 601 (Oct 24)
- What’s New is new ;) We wanted to make Blink updates more discoverable, giving new features a proper place to shine or a way to let you know that we fixed that annoying bug. When Blink updates, it will tell you so in a new shell, and ask you to run the command `whatsnew`.
- Fixes to environment variables ordering.

## 16.0.1 Build 614 (Oct 26)
- Looks like the way windows work in the new OS is a little bit... confusing. To make things more predictable, we have decided to go back to a good old sheet, at least for now.
https://github.com/blinksh/blink/issues/1604

- Some Blink Plus users did not have access to our new Passkeys and Security Keys feature. This should fix it.
https://github.com/blinksh/blink/issues/1602

## 16.0.2 Build 622 (Nov 14)
- Fixed smart keys with different zoom modes #1416
- Fixed crash with external display #1621
- Fixed VS Code folder display #1625
- Fixed Files.app and Code with ProxyCommand #1612 

## 16.0.3 Build 637 (Dec 2)
Big news on this minor version, we updated to the new Mosh 1.4.0. Please make sure to test your setups thoroughly. Here is what else is new:

- Mosh predict overwrite #XXXX
- As we updated Mosh, we made some tweaks for speed and reliability.
- Fixes on Files.app and Code with multiple ProxyJumps. #1390
- Fixes on Files.app leaking storage when deleting elements. #1619
- Fixes some file types greyed out on Files.app #1645
- Files.app now erases the cache when removing the domain. #XXXX
- Fixed Caret on VS Code. #XXXX
- Fixes from crashes on reports. 

## Known Issues
- [WebAuthn keys will not work with SSH Agent](https://github.com/blinksh/blink/discussions/1572).
- WebAuthn keys do not support interactive flows like Files.app or Code integrations. If you think this may be important for you, [please vote here](https://github.com/blinksh/blink/discussions/1573).
- [WebAuthn keys do not synchronize between your devices yet](https://github.com/blinksh/blink/issues/1574)

**Comment by cro** [comment]
Sorry if I'm being dense, does the WebAuthn support only work under iOS 16?

I've never installed a beta iOS on my iPad, but if it works only under 16 that might push me over the edge. 😄

**Reply by yury** [reply]
I don't recommend  to jump on iPadOS 16 yet.

**Comment by Prometheia** [comment]
This isn’t able to be installed unless I have iPadOS 16 installed. The non-beta version of iPadOS 16 won’t be out in time before the previous Blink beta expires. Will there be an update for people still on iPadOS 15 for the time being?

**Reply by carloscabanero** [reply]
We released an update for 15 too. We are thinking to make 16 a hard update though, if Apple figures things out.

**Comment by s00500** [comment]
hm.... so not entirely sure I got how this should work...

What  I did: add a new key using a my yubikey, copy public key from yubikey to server, config host in blink to use this key, start connection

Then I still get asked for a password.... anything wrong in my workflow or shall I retry / double check everything ?

**Reply by yury** [reply]
We released buggy version, please update to build 575.
But you need to create new key after update.

**Reply by s00500** [reply]
Ah I was already at that version, found that the public key copy button in the app adds linebreaks to the key, not a big issue but kinda awkward... works now :-)

**Reply by yury** [reply]
Hmm... will look. Thank you.

**Reply by yury** [reply]
Can't reproduce this yet. Can you help me with that? May be line breaks appear on paste?

**Reply by s00500** [reply]
I am on an iphone, when I copy and then either paste in the notes app or in over on my mac I get 2 linebreaks seperating keytype key and identifer


Like so: 

sk-ecdsa-sha2-nistp256@openssh.com <ln>
Key <ln>
lbsphone@iphone

**Reply by s00500** [reply]
Using the copy button on the key info view of blinks config

**Reply by yury** [reply]
Got it. Thanks.

It seems like Notes wrap lines on spaces. If you rotate your phone `lbsphone@iphone` should be on same line as Key.

**Reply by s00500** [reply]
That was not the case, originally I pressed copy and then pasted on my mac into nano, also had the issue there...

**Reply by s00500** [reply]
Ok... I realized the issue happens ONLY on nano, notes is fine (as you explained) and the micro editor as well as textedit are also fine... weird issue

**Comment by s00500** [comment]
Got another question: I have 1 yubikey with NFC and USBC
I would like to use it for my ipad and my iphone, I generate a key on the ipad and it works, then I generate a key on my ipad and add to the server too, I start ssh and get a key prompt, I put in the yubikey and it dissapears and asks me for password... any idea ?

**Reply by carloscabanero** [reply]
Hmmm... when the key prompt is shown and you connect the key, the key should start to blink but the prompt should not disappear. Can you send us a quick video? Thanks!

**Reply by s00500** [reply]
Ok, riddle is solved... the icloud sync between the two devices only wokrs for the hosts and not the key, so then it seems to unlink them in some way... it works when I name the keys different on the devices and make sure that I have 2 different host configs too, only then they do not override each other

Not a big problem, but a bit funkey :-)
Sent a video on testflight anyways

**Comment by carloscabanero** [comment]
Just released a new Build, 591. We need your help with StageManager and our rebumped Smart Keys bar. Let us know what you think!

**Comment by carloscabanero** [comment]
Build 595 out.

**Comment by carloscabanero** [comment]
Build 601 and 606 are 16.0.0. Both in review for AppStore and TestFlight. Thanks everyone!!

**Comment by aivanise** [comment]
Stupid question, how does one exit the 'whatsnew' screen? I can't see any way out, swiping up or even rebooting does not help, every time I start the app i end up on the whatsnew. Blink 16.0.0 from app store, ipados 15.7, ipad mini 5th gen.

**Reply by yury** [reply]
tap 3 dots at the top and select close. 
Also you can drag up new blink window from dock.

**Reply by aivanise** [reply]
Thanks, i don't have a close option in three dots menu, but I've managed to set it to half screen and swipe it up. Maybe there should be a close button on there for us old school people who still haven't accepted that ipad is a multitasked device, I only ever hit the three dots accidentaly :)

**Reply by yury** [reply]
Ok, we will do just modal so you can swipe down to dismiss it.
Will be in 6.0.1 update

**Comment by carloscabanero** [comment]
Updated notes for 16.0.1 Build 614.

**Comment by carloscabanero** [comment]
Updated notes for 16.0.2. Build 622

**Comment by carloscabanero** [comment]
## 16.0.3 Build 637 (Dec 2)
Big news on this minor version, we updated to the new Mosh 1.4.0. Please make sure to test your setups thoroughly. Here is what else is new:

- Mosh predict overwrite #1034 
- As we updated Mosh, we made some tweaks for speed and reliability.
- Fixes on Files.app and Code with multiple ProxyJumps. #1390
- Fixes on Files.app leaking storage when deleting elements. #1619
- Fixes some file types greyed out on Files.app #1645
- Files.app now erases the cache when removing the domain. #XXXX
- Fixed Caret on VS Code. #1615 
- Fixes from crashes on reports.

**Reply by carloscabanero** [reply]
## 16.0.3 Build 639 (Dec 9)
* Configure keyboard shortcuts for Cmd-W and Cmd-T #1650
